src/Machine.Specifications.Runner.VisualStudio/Execution/TestExecutor.cs

```diff
@@ -59,8 +59,16 @@ public void RunTestsInAssembly(string pathToAssembly, IEnumerable<VisualStudioTe
             }
             finally
             {
-                if (mspecRunner != null && assemblyToRun != null)
-                    mspecRunner.EndRun(assemblyToRun);
+                try
+                {
+                    if (mspecRunner != null && assemblyToRun != null)
+                        mspecRunner.EndRun(assemblyToRun);
+                }
+                catch (Exception exception)
+                {
+                    // TODO Logging
+                    Console.WriteLine(exception);
```

Comment:
What does this need? Also by swallowing the exceptions, isn't this going to cause false positive test cases?

Reply:
We found in the stack trace that if OnFatalError throws, EndRun could also throw.
I copied the IsolatedAppDomainExecutionScope Dispose "TODO".
Not sure the best approach (our tests are happier with this in)

Reply:
@robertcoltheart

Reply:
Ok fair enough. Console writing won't do anything I don't think, maybe this could write to the `IFrameworkHandle` instead?